libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/dialogs/QnAMakerDialog.java

```diff
@@ -665,14 +671,14 @@ public QnAMakerDialog(String sourceFilePath, Integer sourceLineNumber) {
 
             String reply = dc.getContext().getActivity().getText();
             QnAMakerDialogOptions dialogOptions = (QnAMakerDialogOptions) ObjectPath
-                    .getPathValue(dc.getActiveDialog().getState(), OPTIONS);
+                    .getPathValue(dc.getActiveDialog().getState(), OPTIONS, QnAMakerDialogOptions.class);
 
             if (reply.equalsIgnoreCase(dialogOptions.getResponseOptions().getCardNoMatchText())) {
                 // it matches nomatch text, we like that.
                 return CompletableFuture.completedFuture(true);
             }
 
-            List<String> suggestedQuestions = (List<String>) dc.getState().getValue("this.suggestedQuestions");
+            List<String> suggestedQuestions = (List<String>) dc.getState().get("this.suggestedQuestions");
```

Comment:
What's the difference? [C#](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog.cs#L423) uses `getValue()`

Reply:
There aren't differences if I'm not wrong, the functionality is the same